src/management-system-v2/app/(dashboard)/[environmentId]/layout-client.tsx

```diff
@@ -58,7 +59,31 @@ const Layout: FC<
     (item) => !(breakpoint.xs && item && 'type' in item && item.type === 'divider'),
   );
 
-  const menu = <Menu theme="light" mode="inline" items={layoutMenuItems} />;
+  let activeKey: string | undefined;
+  for (const layoutItem of layoutMenuItems) {
+    if (
+      layoutItem === null ||
+      !('type' in layoutItem) ||
+      layoutItem.type !== 'group' ||
+      !layoutItem.children
+    )
+      continue;
+
+    activeKey = layoutItem?.children.find((item) => {
+      return item?.key && pathname.includes(item.key as string);
+    })?.key as string;
+
+    if (activeKey) break;
+  }
```

Comment:
This could fail if we have a menu item like `/executions` as well as `/executions/monitoring` or sth. like that. If you want to do it that way, the check should be 1:1. Alternatively a context could be set through a `Content` component prop that gets read here, so we aren't limited by the url and still set it through the individual pages.

Reply:
I do think the best way is through some form of url mathing, otherwise it would become to annoying to have to set a context value every time, and also there could be some issues with intercepting routes. What do you think of checking that the url ends in a string? This could also cause conflicts, but it is far less likely.

Reply:
Could you explain that more, what do you mean with ends with a string? How would you make sure `/executions/list` or whatever could be at the end marks the `/executions` list item in that case?

Reply:
if the url is ".../executions/list", then `url.endsWith("/executions")` would be false, and `url.endsWith("/executions/list")`  would be true

Reply:
Okay I misunderstood then maybe. I though the intention was to always mark an item, even on sub-sites (like only executions is in the menu, so `/executions/list` would still mark executions item in menu).